backend/app/main.py

```diff
@@ -0,0 +1,158 @@
+    server_metadata_url=(conf.OIDC_PROVIDER_URL + "/.well-known/openid-configuration"),
+    client_kwargs={
+        "scope": conf.OIDC_SCOPES,
+        "timeout": Timeout(timeout=conf.OIDC_REQUEST_TIMEOUT),
+    },
+)
+
+key_fetcher = AsyncKeyFetcher()
+
+
+@router.get("/login")
+async def login(request: Request):
+    return await oauth.login_portal.authorize_redirect(
+        request=request,
+        redirect_uri=f"{conf.BASE_URL}/api/auth",
+        acr_values=conf.OIDC_ACR_VALUES,
+    )
+
+
+@router.get("/auth")
+async def auth(request: Request):
+    try:
+        token = await oauth.login_portal.authorize_access_token(request)
+    except OAuthError as error:
+        raise HTTPException(status_code=401, detail=error.error)
+
+    id_token = token["id_token"]
+    access_token = token["access_token"]
+    expires_in = token["expires_in"]
+
+    response = RedirectResponse(url=conf.BASE_URL)
+
+    if id_token:
+        response.set_cookie(
+            key="id_token",
+            value=id_token,
+            max_age=expires_in,
+            httponly=True,
+        )
+    if access_token:
+        response.set_cookie(
+            key="access_token",
+            value=access_token,
+            max_age=expires_in,
+            httponly=True,
+        )
+
+    return response
+
+
+@router.get("/logout")
+async def logout(id_token: Optional[str] = Cookie(default=None)):
+    metadata = await oauth.login_portal.load_server_metadata()
+    end_session_endpoint = metadata["end_session_endpoint"]
+
+    end_session_uri = add_params_to_uri(
+        end_session_endpoint,
+        (
+            ("id_token_hint", id_token),
+            ("post_logout_redirect_uri", conf.BASE_URL),
+        ),
+    )
+
+    response = RedirectResponse(url=end_session_uri)
+
+    response.delete_cookie(
+        key="id_token",
+        httponly=True,
+    )
+    response.delete_cookie(
+        key="access_token",
+        httponly=True,
+    )
+
+    return response
+
+
+@router.get("/me")
+async def user_profile(id_token: Optional[str] = Cookie(default=None)):
+    """
+    Return information about the currently authenticated user
+    """
+    if id_token:
+        key_entry = await key_fetcher.get_key(id_token)
+        token = jwt.decode(
+            id_token,
+            audience=conf.OIDC_CLIENT_ID,
+            **key_entry,
+        )
+        return {
+            "loggedIn": True,
+            "email": token["email"],
+        }
+    return {
+        "loggedIn": False,
+    }
+
+
+@router.post("/data-product/{data_product:path}")
+async def fetch_data_product(
+    data_product: str,
+    request: Request,
+    source=Query(),
+    id_token: Optional[str] = Cookie(default=None),
+):
+    """
+    Simple proxy from frontend to Product Gateway.
+    Sometimes you have to include secrets to Product Gateway requests,
+    so it's more secure to communicate with the service using a backend
+    and not expose e.g private tokens to public.
```

Comment:
How about something more like this?

```suggestion
    Simple proxy from frontend to Product Gateway.

    Some future requests to the Product Gateway will require authentication 
    of the application, thus we route all the request through the backend.
```